src/Lecture1.hs

```diff
@@ -94,7 +104,11 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str
+  | end <  0 = ""
+  | otherwise = take (end-start2+1) . drop start2 $ str
+    where start2 = if start < 0 then 0 else start
```

Comment:
You can implement this a bit shorter using the standard `max` function
```suggestion
    where start2 = max 0 start
```

Reply:
updated